compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -47,36 +47,27 @@ impl<'a, 'tcx> FindHirNodeVisitor<'a, 'tcx> {
             .infcx
             .in_progress_typeck_results
             .and_then(|typeck_results| typeck_results.borrow().node_type_opt(hir_id));
-        match ty_opt {
-            Some(ty) => {
-                let ty = self.infcx.resolve_vars_if_possible(ty);
-                if ty.walk().any(|inner| {
-                    inner == self.target
-                        || match (inner.unpack(), self.target.unpack()) {
-                            (GenericArgKind::Type(inner_ty), GenericArgKind::Type(target_ty)) => {
-                                match (inner_ty.kind(), target_ty.kind()) {
-                                    (
-                                        &ty::Infer(ty::TyVar(a_vid)),
-                                        &ty::Infer(ty::TyVar(b_vid)),
-                                    ) => self
-                                        .infcx
-                                        .inner
-                                        .borrow_mut()
-                                        .type_variables()
-                                        .sub_unified(a_vid, b_vid),
-                                    _ => false,
-                                }
+        ty_opt.filter(|&ty| {
```

Comment:
Maybe remove the `let` binding now that this uses method calls instead of a match?

Reply:
Heh, good catch. That also happens to reduce the diff by neutralizing the whitespace changes.